orttraining/orttraining/python/training/ortmodule/_training_manager.py

```diff
@@ -45,7 +46,15 @@ def execution_session_run_forward(execution_session, onnx_model, device, *inputs
         forward_outputs = C.OrtValueVector()
         # Run and return module outputs.
         execution_session.run_forward(forward_inputs, forward_outputs, state)
-        user_outputs = tuple(_utils._ortvalue_to_torch_tensor(forward_output) for forward_output in forward_outputs)
+
+        user_outputs = []
+        for i in range(len(forward_outputs)):
+            if i < cache_start:
+                user_outputs.append(_utils._ortvalue_to_torch_tensor(forward_outputs[i]))
+            elif cache !=None:
+                cache.insert(cache_names[i-cache_start], forward_outputs[i])
+
+        user_outputs = tuple(user_outputs)
```

Comment:
>user_outputs = tuple(user_outputs) [](http://example.com/codeflow?start=8&length=34)

Can you check this won't result in a copy?

Reply:
good point, avoiding the use.